File: Examples/DotNet/CSharp/Pictures/ConvertPictureToImage.cs
﻿using System;
using Com.Aspose.Cells.Api;
using Com.Aspose.Cells.Model;
using Com.Aspose.Storage.Api;

namespace Pictures
{
    class ConvertPictureToImage
    {
        public static void Run()
        {
            // ExStart:1
            CellsApi cellsApi = new CellsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String name = "Sample_Test_Book";
            String fileName = name + ".xls";
            String sheetName = "Sheet6";
            int pictureNumber = 0;
            String format = "png";
            String storage = "";
            String folder = "";

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));

                // Invoke Aspose.Cells Cloud SDK API to convert picture to image
                ResponseMessage apiResponse = cellsApi.GetWorksheetPictureWithFormat(fileName, sheetName, pictureNumber, format, storage, folder);

                if (apiResponse != null)
                {
                    Console.WriteLine("Convert Picture to Image, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Node.js/SDK/Pictures/ConvertPictureToImage.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var CellsApi = require('asposecellscloud');
var configProps = require('../Data/config.json');
var data_path = '../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Cells API SDK
var cellsApi = new CellsApi(config);

// Set input file name
var filename = "Sample_Test_Book";
var name = filename + ".xls";
var sheetName = "Sheet6";
var pictureNumber = 0;
var format = "png";
try {
// Upload source file to aspose cloud storage
	storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
		assert.equal(responseMessage.status, 'OK');
		
		// Invoke Aspose.Cells Cloud SDK API to convert a picture to image
		cellsApi.GetWorksheetPictureWithFormat(name, sheetName, pictureNumber, format, null, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			
			// Get converted image from response stream
			var outfilename = filename + pictureNumber + "." + format;
			var writeStream = fs.createWriteStream(data_path + "_out_" + outfilename);
			writeStream.write(responseMessage.body);

			});

		});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1